api/src/main/java/io/minio/MinioClient.java

```diff
@@ -903,6 +903,12 @@ private Request createRequest(Method method, String bucketName, String objectNam
       for (String pathSegment : objectName.split("/")) {
         urlBuilder.addEncodedPathSegment(S3Escaper.encode(pathSegment));
       }
+      if (objectName.endsWith("/")) {
+        // When objectName ends with "/", we add a path segment / at the end of URL. 
+        // This is to comply with aws-sdk-java behavior 
+        // See https://github.com/minio/minio-java/issues/648 for more details
```

Comment:
Below comment is good enough

```
// Fix issue #648: preserve tailing '/' to work with proxies like nginx.
```

Reply:
:+1:

Reply:
trailing '/' is better

Reply:
@poornas I have updated the comment as `// Fix issue #648: preserve tailing '/' to work with proxies like nginx.`

Reply:
@poornas I got your point now :)